keylime/tpm/tpm_main.py

```diff
@@ -1200,6 +1273,48 @@ def check_quote(
             hash_alg,
         )
 
+    def check_quote_timing(self, previous_clockinfo, quote):
+        # Sanity check quote clock information
+
+        current_clockinfo = {}
+
+        if previous_clockinfo:
+
+            pclki = TPMClockInfo.from_json(previous_clockinfo)
+
+            retout, success = self._tpm2_printquote(quote, False)
+            if not success:
+                return "tpm2_print failed with " + str(retout), current_clockinfo
```

Comment:
You can also construct a failure object and return that. This is more expressive than a string.

Reply:
Not sure, will keep it as is for now